internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -450,7 +467,11 @@ func (r *BMACReconciler) reconcileBMH(ctx context.Context, bmh *bmh_v1alpha1.Bar
 			// are done at the beginning of this function.
 			bmh.Spec.Image = &bmh_v1alpha1.Image{}
 			liveIso := "live-iso"
-			bmh.Spec.Image.URL = infraEnv.Status.ISODownloadURL
+			u, err := httpISODownloadURL(infraEnv.Status.ISODownloadURL, infraEnv.Namespace)
```

Comment:
I'm making an assumption here that the service is running in the same namespace as where the infraenv exists.

Based on the env I've been looking at today from @rwsu that doesn't seem to always be true.
Does that sound right to you @rwsu @flaper87 ? I think we have access to the namespace through an env var so I can change to use that instead.

Reply:
I just went ahead and changed it to something more reliable.